arelle/UrlUtil.py

```diff
@@ -364,6 +364,10 @@ def isAbsolute(url: str) -> bool:
 def isHttpUrl(url: str) -> bool:
     return isinstance(url,str) and (url.startswith("http://") or url.startswith("https://"))
 
+opaqueUriPattern = re.compile(r'[^:/?#]+:[^/]')
+def isOpaque(url: str) -> bool:
+    return opaqueUriPattern.match(url) is not None
```

Comment:
Ah, [RFC 3986 section 5.2.2](https://datatracker.ietf.org/doc/html/rfc3986#section-5.2.2) has this snippet (I think I overlooked that due to the page break):
```
      -- A non-strict parser may ignore a scheme in the reference
      -- if it is identical to the base URI's scheme.
      --
      if ((not strict) and (R.scheme == Base.scheme)) then
         undefine(R.scheme);
      endif;
```
...and also in [section 5.4.2](https://datatracker.ietf.org/doc/html/rfc3986#section-5.4.2):
```
   Some parsers allow the scheme name to be present in a relative
   reference if it is the same as the base URI scheme.  This is
   considered to be a loophole in prior specifications of partial URI
   [RFC1630].  Its use should be avoided but is allowed for backward
   compatibility.

      "http:g"        =  "http:g"         ; for strict parsers
                      /  "http://a/b/c/g" ; for backward compatibility
```
That's a mess for offline processing of things that would eventually be hosted on `http`/`https`.  It seems like it would be reasonable for iXBRL processing to be a "strict" processor and use your patch as-is.